app/models/miq_expression/field.rb

```diff
@@ -53,8 +63,58 @@ def does_not_match(other)
     arel_attribute.does_not_match(other, escape, case_sensitive)
   end
 
+  def contains(other)
+    raise unless associations.one?
+    reflection = reflections.first
+    arel = eq(other)
+    arel = arel.and(Arel::Nodes::SqlLiteral.new(extract_where_values(reflection.klass, reflection.scope))) if reflection.scope
+    model.arel_attribute(:id).in(
+      target.arel_table.where(arel).project(target.arel_table[reflection.foreign_key]).distinct
+    )
+  end
+
   private
 
+  class WhereExtractionVisitor < Arel::Visitors::PostgreSQL
```

Comment:
@kbrock yeah, I have only moved and fixed this code ;). I think it was a refactoring that Matthew made to better handle the CONTAINS logic that was going on before Basically makes sure you have all the right joins as defined by the model's relationships. I think it is still needed, though as a side note it wouldn't be if we were using the AR API